packages/framework/src/Model/Order/PromoCode/PromoCodeTypeEnum.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Order\PromoCode;
+
+use Shopsys\FrameworkBundle\Component\Enum\AbstractEnum;
+
+class PromoCodeTypeEnum extends AbstractEnum
```

Comment:
I know you will probably hate me soon, however, we should also create `PromoCodeTypeEnum(+Decorator)` in FE API and use it in `PromoCodeDecorator.type` :upside_down_face: